internal/gitprovider/gitea/gitea.go

```diff
@@ -0,0 +1,286 @@
+package gitea
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"code.gitea.io/sdk/gitea"
+
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/gitprovider"
+)
+
+const ProviderName = "gitea"
+
+var registration = gitprovider.Registration{
+	Predicate: func(repoURL string) bool {
+		u, err := url.Parse(repoURL)
+		if err != nil {
+			return false
+		}
+		// We assume that any hostname with the word "gitea" in it can use this
+		// provider
+		return strings.Contains(u.Host, ProviderName)
+	},
+	NewProvider: func(
+		repoURL string,
+		opts *gitprovider.Options,
+	) (gitprovider.Interface, error) {
+		return NewProvider(repoURL, opts)
+	},
+}
+
+func init() {
+	gitprovider.Register(ProviderName, registration)
+}
+
+type giteaClient interface {
+	CreatePullRequest(
+		ctx context.Context,
+		owner string,
+		repo string,
+		opts *gitea.CreatePullRequestOption,
+	) (*gitea.PullRequest, *gitea.Response, error)
+
+	ListPullRequests(
+		ctx context.Context,
+		owner string,
+		repo string,
+		opts *gitea.ListPullRequestsOptions,
+	) ([]*gitea.PullRequest, *gitea.Response, error)
+
+	GetPullRequests(
+		ctx context.Context,
+		owner string,
+		repo string,
+		number int,
+	) (*gitea.PullRequest, *gitea.Response, error)
+
+	AddLabelsToIssue(
+		ctx context.Context,
+		owner string,
+		repo string,
+		number int,
+		labels []string,
+	) ([]*gitea.Label, *gitea.Response, error)
+}
+
+// provider is a Gitea implementation of gitprovider.Interface.
+type provider struct { // nolint: revive
+	owner  string
+	repo   string
+	client giteaClient
+}
+
+// NewProvider returns a Gitea-based implementation of gitprovider.Interface.
+func NewProvider(
+	repoURL string,
+	opts *gitprovider.Options,
+) (gitprovider.Interface, error) {
+	if opts == nil {
+		opts = &gitprovider.Options{}
+	}
+
+	host, owner, repo, err := parseRepoURL(repoURL)
+	if err != nil {
+		return nil, err
+	}
+
+	var clientOpts []gitea.ClientOption
+	if opts.Token != "" {
+		clientOpts = append(clientOpts, gitea.SetToken(opts.Token))
+	}
+	if opts.InsecureSkipTLSVerify {
+		clientOpts = append(clientOpts, gitea.SetHTTPClient(&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}))
+	}
+
+	baseURL := "https://" + host
```

Comment:
This needs a double check @krancour. I "enforced" HTTPS here based on what we do for GitHub, but I am not sure that's what we want to do.

Reply:
I started to look at that a few days ago and got sidetracked. I'll assign myself to this so that I hopefully don't lose track.

Sorry for the delay.